### ISSUE TITLE
Copying png (binary) files is modifying the content

### ISSUE THREAD

**Issue by Jako** [gulpjs/gulp#2766]
Before you open this issue, please complete the following tasks:

* [x] use the search bar at the top of the page to search this repository for similar issues or discussions that have already been opened.
* [x] if you think there is a bug in our code, open this issue.

### Issue

After updating to Gulp 5, the content of png files is modified, when the files are copied with src and dest. This was working fine with Gulp 4.

### What were you expecting to happen?

Not modified files.

### What actually happened?

Part of the png content before:

```
89504E47 0D0A1A0A 0000000D 49484452 00000148 0000005A 08060000 00DA7DCF 1700001C 79494441 547801EC DA038C34 0718C6F1 DA6D5CDB B6CEB66D DBBEA8B6 ADA0B66D F77CB7B6 CEB68DB7 6FCDCF33 7BB3C933 C9EFCC27 C97F31B3 9F940E1C 383A4C3E 17B22C76 3F7B8735 313DEBFC 9D96FDC4 DE6477B1 647626DB 4F680018 01A4C08B 3DCDBA18 ED253DBB 8F5DE5E8 7B000209 7024ABFA 238A0253
```

The copied result:

```
EFBFBD50 4E470D0A 1A0A0000 000D4948 44520000 01480000 005A0806 000000EF BFBD7DEF BFBD1700 001C7949 44415478 01EFBFBD EFBFBD03 EFBFBD34 0718EFBF BDEFBFBD EFBFBD6D 5CDBB6CE B66DDBBE EFBFBDEF BFBDEFBF BDEFBFBD EFBFBD6D EFBFBD7C EFBFBDEF BFBDCEB6 EFBFBDEF BFBD6FEF BFBDEFBF BD337BEF BFBDEFBF BD33EFBF BDEFBFBD EFBFBD27 EFBFBD7F 31EFBFBD EFBFBDEF BFBD0E1C
```

It looks like the content is encoded to UTF8 during copying the files. Can this be disabled somehow?

### Please give us a sample of your gulpfile

The task is quite simple:

```js
const imagesMgr = function () {
    return gulp.src('./src/img/**/*.+(png|jpg|gif|svg)')
        .pipe(gulp.dest('dest/img/'));
};
gulp.task('images', gulp.series(imagesMgr));
```

### Please provide the following information:
* OS & version [e.g. MacOS Catalina 10.15.4]: 14.4.1
* node version (run `node -v`): v18.18.2
* npm version (run `npm -v`): 10.2.5
* gulp version (run `gulp -v`): 5.0.0

**Comment by phated** [comment]
This is by design and the first breaking change highlighted in our release blog: https://medium.com/gulpjs/announcing-gulp-v5-c67d077dbdb7

**Comment by Jako** [comment]
How can this be avoided? Or is copying files not wanted anymore?

**Comment by phated** [comment]
If you are using src just to glob and then copy files, you should have always been using { read: false } because loading files into memory is way slower than a copy. This is a problem with how you use gulp, not the migration.

**Comment by Jako** [comment]
There are a lot of examples in the web using just the src/dest combination without { read: false }. So it will be much easier for people who are not so much into the details of Gulp to repair the issues with their gulp scripts with a migration documentation.

The docs of src() don't contain the new encoding option. And the docs of the read option is a bit misleading. The result is somehow writable to disk via dest().

**Comment by phated** [comment]
Yep, a lot of examples are bad 👍 We have an open issue to update the docs. Feel free to contribute.

**Comment by Jako** [comment]
Writing documentation should not be placed primary on the shoulders of users. This is a primary task of the programmers and managers of a project.

And now let's see how many duplicates were linked with this issue in the next weeks.

**Comment by phated** [comment]
@Jako you are sorely mistaken on how open source works. Maintainers owe you nothing. I've marked your comment as spam and locked this thread. Further outbursts will result in a ban.

**Comment by yocontra** [comment]
Docs have been updated: https://github.com/gulpjs/gulp/blob/master/docs/api/src.md#options

You need to add `{ encoding: false }` for binary files.
